### PR TITLE
IGNITE-19225 Generate beginTs on a coordinator and send it to all tx participants

### PR DESCRIPTION
**Pull request by rpuch**
https://issues.apache.org/jira/browse/IGNITE-19225

### CODE REVIEW

File: modules/table/src/test/java/org/apache/ignite/internal/table/distributed/command/PartitionRaftCommandsSerializationTest.java [hunk 1]
```diff
@@ -85,7 +85,7 @@ public void testUpdateCommand() throws Exception {
                         .partitionId(1)
                         .build()
                 )
-                .rowUuid(Timestamp.nextVersion().toUuid())
+                .rowUuid(TestTransactionIds.newTransactionId())
                 .rowBuffer(byteBufferFromBinaryRow(1))
                 .txId(UUID.randomUUID())
```
Comment (SammyVimes) [inline modules/table/src/test/java/org/apache/ignite/internal/table/distributed/command/PartitionRaftCommandsSerializationTest.java:90]: why `TestTransactionIds.newTransactionId()` not here but at `rowUuid`?

File: modules/transactions/src/testFixtures/java/org/apache/ignite/internal/tx/test/TestTransactionIds.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+/**
+ * Contains methods to generate transaction IDs for the cases when the clock and nodeId might be hard-coded
+ * (just for tests).
+ */
+public class TestTransactionIds {
+    /** Global clock instance. */
+    private static final HybridClock GLOBAL_CLOCK = new HybridClockImpl();
+
+    /** Hard-coded node ID. */
+    private static final int SOLE_NODE_ID = 0xdeadbeef;
```
Comment (SammyVimes) [inline modules/transactions/src/testFixtures/java/org/apache/ignite/internal/tx/test/TestTransactionIds.java:34]: Can you use `TransactionIdGenerator` here? This class feels like a copy of it

**Review by SammyVimes — CHANGES_REQUESTED** [review]

File: modules/transactions/src/testFixtures/java/org/apache/ignite/internal/tx/test/TestTransactionIds.java (hunk 2 above)
Comment (rpuch) [reply modules/transactions/src/testFixtures/java/org/apache/ignite/internal/tx/test/TestTransactionIds.java:34]: Not sure if this is what you ment, let's see. The thing is that `TransactionIdGenerator` only encapsulates nodeId, not the clock. Maybe not the best name for the abstraction.

File: modules/table/src/integrationTest/java/org/apache/ignite/internal/table/ItColocationTest.java [hunk 3]
```diff
@@ -213,7 +219,7 @@ public CompletableFuture<Void> finish(
                                         .partitionId(commitPartId.partitionId())
                                         .build()
                         )
-                        .rowUuid(Timestamp.nextVersion().toUuid())
+                        .rowUuid(TestTransactionIds.newTransactionId())
```
Comment (SammyVimes) [inline modules/table/src/integrationTest/java/org/apache/ignite/internal/table/ItColocationTest.java:222]: same about `txId`

File: modules/table/src/test/java/org/apache/ignite/internal/table/distributed/command/PartitionRaftCommandsSerializationTest.java [hunk 4]
```diff
@@ -105,7 +105,7 @@ public void testRemoveCommand() throws Exception {
                         .partitionId(1)
                         .build()
                 )
-                .rowUuid(Timestamp.nextVersion().toUuid())
+                .rowUuid(TestTransactionIds.newTransactionId())
```
Comment (SammyVimes) [inline modules/table/src/test/java/org/apache/ignite/internal/table/distributed/command/PartitionRaftCommandsSerializationTest.java:108]: same about `txId`

File: modules/runner/src/main/java/org/apache/ignite/internal/app/IgniteImpl.java [hunk 5]
```diff
@@ -352,7 +353,8 @@ public class IgniteImpl implements Ignite {
 
         ReplicaService replicaSvc = new ReplicaService(clusterSvc.messagingService(), clock);
 
-        txManager = new TxManagerImpl(replicaSvc, lockMgr, clock);
+        // TODO: IGNITE-19344 - use nodeId that is validated on join (and probably generated differently).
+        txManager = new TxManagerImpl(replicaSvc, lockMgr, clock, new TransactionIdGenerator(() -> clusterSvc.nodeName().hashCode()));
```
[the remark this answers is not in the document]
Comment (rpuch) [reply modules/runner/src/main/java/org/apache/ignite/internal/app/IgniteImpl.java:357]: `nodeId` mentioned in the TODO does not yet exist, it should be a 32-bit number, and the mentioned ticket is about adding this `nodeId` (maybe the name is not perfect).

**Review by SammyVimes — CHANGES_REQUESTED** [review]

**Review by SammyVimes — APPROVED** [review]
LGTM

**Pull request merged**